crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -96,6 +96,10 @@ def health_check
         ret[:nodes_not_ready] = unready unless unready.empty?
         failed = Proposal.all.select { |p| p.active? && p.failed? }
         ret[:failed_proposals] = failed.map(&:display_name) unless failed.empty?
+        # Not using Node.all here as we only need node names
+        all_node_names = ChefObject.fetch_nodes_from_cdb.first.map(&:name)
+        unsaved = Proposal.all.reject { |p| (p.all_elements - all_node_names).empty? }
```

Comment:
Can't we actually save the result of previous Proposal.all so we don't have to do the same search again?

Reply:
done